proparse/src/main/java/org/prorefactor/proparse/antlr4/ProgressLexer.java

```diff
@@ -408,16 +492,16 @@ private void getRawChar() throws IOException {
             throw new ProparseRuntimeException("Proparse error. Infinite loop caught by preprocessor.");
           return;
         case 1: // popped an include file
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1151)